File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }

        int i = 0, j = 0;
        sort(s.begin(), s.end());
        sort(t.begin(), t.end());

        while(i<s.size() && j<t.size()){
            if(s[i] != t[j]){
                return false;
            }
            i++;
            j++;
        }

        return true;
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> smap = new HashMap<>();
        Map<Character, Integer> tmap = new HashMap<>();

        for(int i = 0; i<s.length(); i++){
            smap.put(s.charAt(i), smap.getOrDefault(s.charAt(i), 0) + 1);
        }

        for(int i = 0; i<t.length(); i++){
            tmap.put(t.charAt(i), tmap.getOrDefault(t.charAt(i), 0) + 1);
        }

        return smap.equals(tmap);
    }
}